core/pom.xml

```diff
@@ -30,8 +30,8 @@ SPDX-License-Identifier: LGPL-3.0-or-later
       <artifactId>smallrye-common-constraint</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.glassfish</groupId>
-      <artifactId>jakarta.json</artifactId>
+      <groupId>jakarta.json</groupId>
+      <artifactId>jakarta.json-api</artifactId>
```

Comment:
I believe the modern replacement for `org.glassfish:jakarta.json` is `org.eclipse.parsson:jakarta.json`.  Currently the Quarkus BOM points to `org.eclipse.parsson:parsson`, so that's what we ought to use.